internal/service/kendra/index.go

```diff
@@ -0,0 +1,1036 @@
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(indexEditionValues(types.IndexEdition("").Values()...), false),
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					// API returns "ENTERPRISE_EDITION" by default.
+					if old == string(types.IndexEditionEnterpriseEdition) && new == "" {
+						return true
+					}
+
+					return old == new
+				},
+			},
+			"error_message": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"index_statistics": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"faq_statistics": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"indexed_question_answers_count": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"text_document_statistics": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"indexed_text_bytes": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"indexed_text_documents_count": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 1000),
+					validation.StringMatch(
+						regexp.MustCompile(`[a-zA-Z0-9][a-zA-Z0-9_-]*`),
+						"The name must consist of alphanumerics, hyphens or underscores.",
+					),
+				),
+			},
+			"role_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+			"server_side_encryption_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"kms_key_id": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							ValidateFunc: validation.StringLenBetween(1, 2048),
+						},
+					},
+				},
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"user_context_policy": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice(userContextPolicyValues(types.UserContextPolicy("").Values()...), false),
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					// API returns "ATTRIBUTE_FILTER" by default.
+					if old == string(types.UserContextPolicyAttributeFilter) && new == "" {
+						return true
+					}
+
+					return old == new
+				},
```

Comment:
maybe consider adding `Default: string(types.UserContextPolicyAttributeFilter),` here instead?

Reply:
Resolved in 22c934249a7774416a04803e8938d509c82c192f